main.go

```diff
@@ -208,6 +212,16 @@ func main() {
 		}
 	}
 
+	if startK8sWatcher != nil && *startK8sWatcher {
+		if apiShutdown, err := cmd.StartK8sWatcher(logger); err != nil {
```

Comment:
I think the execution should stop if the service is not running. Otherwise it will be hard to troubleshoot if it's ignored.

Reply:
You mean the service should always be running?
For local execution in one instance this will not be needed. It is only needed when running a decentralised storage and ARC running in a K8s cluster.